include/boost/geometry/strategies/cartesian/side_robust.hpp

```diff
@@ -0,0 +1,190 @@
+
+#include <boost/geometry/util/select_most_precise.hpp>
+#include <boost/geometry/util/select_calculation_type.hpp>
+#include <boost/geometry/util/precise_math.hpp>
+#include <boost/geometry/util/math.hpp>
+
+namespace boost { namespace geometry
+{
+
+namespace strategy { namespace side
+{
+
+struct eps_equals_policy
+{
+public:
+
+    template <typename Policy, typename T1, typename T2>
+    static auto apply(T1 a, T2 b, Policy policy)
+    {
+        return boost::geometry::math::detail::equals_by_policy(a, b, policy);
+    }
+};
+
+struct fp_equals_policy
+{
+public:
+    template <typename Policy, typename T1, typename T2>
+    static auto apply(T1 a, T2 b, Policy)
+    {
+        return a == b;
+    }
+};
+
+
+/*!
+\brief Adaptive precision predicate to check at which side of a segment a point lies:
+    left of segment (>0), right of segment (< 0), on segment (0).
+\ingroup strategies
+\tparam CalculationType \tparam_calculation (numeric_limits<ct>::epsilon() and numeric_limits<ct>::digits must be supported for calculation type ct)
+\tparam Robustness std::size_t value from 0 (fastest) to 3 (default, guarantees correct results).
+\details This predicate determines at which side of a segment a point lies using an algorithm that is adapted from orient2d as described in "Adaptive Precision Floating-Point Arithmetic and Fast Robust Geometric Predicates" by Jonathan Richard Shewchuk ( https://dl.acm.org/citation.cfm?doid=237218.237337 ). More information and copies of the paper can also be found at https://www.cs.cmu.edu/~quake/robust.html . It is designed to be adaptive in the sense that it should be fast for inputs that lead to correct results with plain float operations but robust for inputs that require higher precision arithmetics.
+ */
+template
+<
+    typename CalculationType = void,
+    typename EpsPolicy = eps_equals_policy,
+    std::size_t Robustness = 3
+>
+struct side_robust
+{
+    template <typename Policy>
+    struct eps_policy
+    {
+        eps_policy() {}
+        template <typename Type>
+        eps_policy(Type const& a, Type const& b, Type const& c, Type const& d)
+            : policy(a, b, c, d)
+        {}
+        Policy policy;
+    };
+
+public:
+
+    typedef cartesian_tag cs_tag;
+
+    //! \brief Computes the sign of the CCW triangle p1, p2, p
+    template
+    <
+        typename PromotedType,
+        typename P1,
+        typename P2,
+        typename P,
+        typename EpsPolicyInternal,
+        std::enable_if_t<std::is_fundamental<PromotedType>::value, int> = 0
+    >
+    static inline PromotedType side_value(P1 const& p1,
+                                          P2 const& p2,
+                                          P const& p,
+                                          EpsPolicyInternal& eps_policy)
+    {
+        using vec2d = ::boost::geometry::detail::precise_math::vec2d<PromotedType>;
+        vec2d pa;
+        pa.x = get<0>(p1);
+        pa.y = get<1>(p1);
```

Comment:
Right this was failing to compile before. I was working on a PR fixing that. There are several possibilities how this could be fixed:
1. assigning values after creation like you did
2. defining a constructor/constructors in `vec2d` and replacing '{}' with '()'
3. using `numeric_cast<PromotedType>(...)`
In my PR I decided to use `numeric_cast` because this also silences conversion warnings with msvc but this may not be desireable. Because maybe there should be warnings. On the other hand these warnings are only generated with msvc (or maybe it depends on compiler options?). Furthermore these warnings are generated in other strategies too so maybe this is a feature. ;)

So anyway, I'm ok with this solution but would like to know your opinion why do you think it's correct?

Reply:
I selected (1) since I found it simpler than (2). Also compared to (3) it will not silence conversion warning that could be useful since we do not test this piece of code with many types (maybe we should) apart from `double` and `long long int` AFAIR. Also (3) is dependent on other `boost` libraries which of course is not a problem but why not an independent solution?